### PR TITLE
Lexer

### PR DESCRIPTION
**Pull request by KauriMai**
Initial Lexer pull request for code Review

### CODE REVIEW

File: src/lex.cpp [hunk 1]
```diff
@@ -1 +1,58 @@
...
+
+    Lexer lex;
+
+    char currCharacter;
+    std::string currentLine;
+    int lineNumber = 1;
+    
+    while (std::cin.get(currCharacter)) {
+
+        if (currCharacter == '\n') {
```
Comment (YanezAndrew) [inline src/lex.cpp:17]: What does this do?

File: src/lib/Lexer.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <cctype>
+
+class Lexer {
+    std::vector<Token> tokens;
+
+    public:
```
Comment (YanezAndrew) [inline src/lib/Lexer.h:11]: Great JOB! I like how simple this .h file is!

File: src/lib/Lexer.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    char currentCharacter;
+    int currCol = 1;
+
+    // if (input.empty()) {
+    //         Token potentialToken(lineNumber,currCol,"END", END);
+    //         tokens.push_back(potentialToken);
+    //         return;
+    // }
+
+    while (transformer.get(currentCharacter)) {
```
Comment (YanezAndrew) [inline src/lib/Lexer.cpp:19]: Summarize what this function does

File: src/lib/Lexer.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            tokens.push_back(newToken);
+            currCol++;
+        }
+
+        else if(isdigit(currentCharacter)){
+            std::string finalNum = std::string(1,currentCharacter);
+            int initCol = currCol;
+            size_t hasDecimal = 0;
+
+            while( isdigit(transformer.peek()) || (transformer.peek() == '.' && hasDecimal == 0) ){
```
Comment (YanezAndrew) [inline src/lib/Lexer.cpp:50]: Say what while loop does.

**Review by YanezAndrew — COMMENTED** [review]
This code was so beautifully written, but a couple of comments about what each loop and if statement would help.

File: src/lib/Lexer.cpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+}
+
+void Lexer::transformLine(int lineNumber, std::string input) {
+    
+    std::istringstream transformer(input);
+    char currentCharacter;
+    int currCol = 1;
+
+    // if (input.empty()) {
```
Comment (perrytham) [inline src/lib/Lexer.cpp:13]: Can you delete the comment?

**Review by perrytham — COMMENTED** [review]
good job briarui

File: src/lex.cpp (hunk 1 above)
Comment (KauriMai) [reply src/lex.cpp:17]: comment created
adds character to currentLine until '\n' is reached

File: src/lib/Lexer.cpp (hunk 5 above)
Comment (KauriMai) [reply src/lib/Lexer.cpp:13]: done

File: src/lib/Lexer.cpp (hunk 4 above)
Comment (KauriMai) [reply src/lib/Lexer.cpp:50]: Created Comment
//Gets all digits until second decimal or space is reached

File: src/lib/Lexer.cpp (hunk 3 above)
Comment (KauriMai) [reply src/lib/Lexer.cpp:19]: Created Comment
//Creates Tokens for Vector until end of line is reached

**Comment by KauriMai** [comment]
Created comments for the loops and logic statements, and also did some code clean-up

**Review by perrytham — APPROVED** [review]
Good job K. B.

**Review by YanezAndrew — APPROVED** [review]
I like how detailed the comments are.

**Pull request merged**